### PR TITLE
📝 `Marketplace`: Updates Stripe docs in README 

### PR DESCRIPTION
**Pull request by rosschapman**
Took a quick-ish stab at expanding the docs to give devs a stronger helping hand when setting up payments for Marketplace. Also note that I created a new `/docs` dir so that doc-related assets could live separately from code. 

What do y'all think?

### CODE REVIEW

**Comment by zspencer** [comment]
It’s a bit hard to see what changed with the move and the edits; mind unwinding the move for now? We normally use README for the “here’s the bare minimum” and docs for breaking out other useful stuff

**Comment by anaulin** [comment]
Just having the move be in a separate commit from the changes would be enough to see more clearly what has changed, I think.

**Review by zspencer — APPROVED** [review]
OK, I read through this and I feel pretty good about it! Two things:

1. I do worry about hand-holding *too much* in our internal docs; by which I mean:

- What parts of the docs are likely to go "out of date" by virtue of us simply improving our UX? (For example, setting the webhook secret via the UI vs via the console)
- What parts are duplicious of Stripe (For example, could the "turn on Test Mode" could be a link to https://stripe.com/docs/test-mode?)

2. Putting the README in the docs means anything that previously link to `<repo>/app./furniture/marketplace` will no longer show the README. It may be worth keeping at least a stub with an Engineering-forward overview of the Marketplace; and then have specific sub-documents for things like `docs/working-with-stripe-on-local-environments.md`

That said, I think this is a substantative improvement; and you should feel free to merge at your leisure!

**Comment by KellyAH** [comment]
> It’s a bit hard to see what changed with the move and the edits

I agree. But it's ok. I have not finished setting up my local dev environment for the Stripe workflow. So this is a great opportunity for me to test out the readme changes.

File: app/furniture/marketplace/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+4. `Shopper` is redirected to the `Order`, which represents the `Cart` to the `Shopper` in a read-only manner.
+
+## Architecture
+
+The Marketplace uses Stripe for online payments. Developers should use the ["Connect then Transfer" workflow](https://stripe.com/docs/connect/collect-then-transfer-guide) as a guide to the overall architectural pattern. At a high level, implementing this pattern boils down to three areas of concern:
+
+1. Build the Workflow for connecting a Stripe Account
+  - Currently implemented for Space administrators with a "Connect to Stripe" button
+2. Checkout with Stripe Checkout, and include the payment_intent_data with a transfer_group: https://stripe.com/docs/api/checkout/sessions/create#create_checkout_session-payment_intent_data-transfer_group
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:68]: It maybe easier to see payment_intent_data and transfer_group if it was quoted.
```suggestion
2. Checkout with Stripe Checkout, and include the `payment_intent_data` with a `transfer_group`: https://stripe.com/docs/api/checkout/sessions/create#create_checkout_session-payment_intent_data-transfer_group
```

File: app/furniture/marketplace/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+For the "Connect to Stripe" flow to work, you will need to have a local development URL that is recognized by Stripe as "reachable". One way to do this is:
+1. Add `127.0.0.1 convene.local` to your `/etc/hosts`
+1. Access the app locally through `convene.local:3000`, instead of `localhost`
+1. Optionally, add `convene.local` as the `branded_domain` to whatever Space you are using for testing, to shorten the URL you use to access it
+
+Next, add your secret API key:
+1. Click on the `Developers` button next to `Test mode` in the top right of your Stripe dasbhoard
+1. Find your `Secret key` and copy it
+1. Paste the key into `Api token` field on your Stripe Utility settings page
+  - http://convene.local:3000/spaces/<space_name>/utilities/<utiltiy_id>/edit
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:108]: is this supposed to render as a partial link to help readers quickly jump to the correct page? Or is it supposed to be quoted like so 👇 ?
```suggestion
  - `http://convene.local:3000/spaces/<space_name>/utilities/<utiltiy_id>/edit`
```

File: app/furniture/marketplace/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+Next, configure Stripe to forward events to a local obtain a webhook secret key via the stripe cli:
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:110]: did you mean?
```suggestion
Next, configure Stripe to forward events to local via obtaining a webhook secret key via the stripe cli:
```

File: app/furniture/marketplace/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+1. Install [`stripe` cli](https://stripe.com/docs/stripe-cli) if you haven't already
+1. Find the `--foward-to` url for your marketplace so by using by printing the value of  `polymorphic_url(marketplace.location(child: :stripe_events))` from within a controller or view debug session
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:113]: did you mean?
```suggestion
1. Find the `--foward-to` url for your marketplace by printing the value of  `polymorphic_url(marketplace.location(child: :stripe_events))` from within a controller or view debug session
```

File: app/furniture/marketplace/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  - Eg, from view: `<%- debug polymorphic_url(marketplace.location(child: :stripe_events)) %>`
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:114]: E.g. is a Latin abbreviation that stands for “exempli gratia,” which translates to “for the sake of example.”
```suggestion
  - E.g., from view: `<%- debug polymorphic_url(marketplace.location(child: :stripe_events)) %>`
```

File: app/furniture/marketplace/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+> Ready! You are using Stripe API Version [2022-11-15]. Your webhook signing secret is whsec_***** (^C to quit)
+```
+1. Update the marketplace settings directly in the database by adding the `stripe_webhook_endpoint_secret` to `public.furnitures.settings`
+
+Finally, let's connect!
+
+1. First, create a Connected account within Stripe by visiting [https://dashboard.stripe.com/test/connect/accounts/overview](https://dashboard.stripe.com/test/connect/accounts/overview) and clicking the `Create` button
+  - You might name this account something like `<Space_Name> Dev`
+  - [Learn more](https://stripe.com/docs/connect)
+1. Follow instructions for creating a `Standard account`
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:128]: Are these the correct instructions?
```suggestion
1. Follow [instructions](https://stripe.com/docs/connect/standard-accounts) for creating a `Standard account`
```

File: app/furniture/marketplace/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+  - Eg, from view: `<%- debug polymorphic_url(marketplace.location(child: :stripe_events)) %>`
+  - [Read more](https://stripe.com/docs/webhooks/test#webhook-test-cli)
+1. Set stripe event forwarding to this url by executing: `$ stripe listen --forward-to <YOUR_FORWARD_TO_URL>`
+1. Copy the webhook signing secret from the output. The output from the above command should look something like:
+```
+> Ready! You are using Stripe API Version [2022-11-15]. Your webhook signing secret is whsec_***** (^C to quit)
+```
+1. Update the marketplace settings directly in the database by adding the `stripe_webhook_endpoint_secret` to `public.furnitures.settings`
+
+Finally, let's connect!
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:123]: I got lost here. what are we connecting? A Standard account in Stripe to convene local?

File: app/furniture/marketplace/docs/README.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+1. First, create a Connected account within Stripe by visiting [https://dashboard.stripe.com/test/connect/accounts/overview](https://dashboard.stripe.com/test/connect/accounts/overview) and clicking the `Create` button
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:125]: I got stuck here. Turns out there are 2 create buttons. `Create v` in the top menu and `+ Create` in the "Connected accounts" section. 

plus create button

```suggestion
1. First, create a Connected account within Stripe by visiting [https://dashboard.stripe.com/test/connect/accounts/overview](https://dashboard.stripe.com/test/connect/accounts/overview) and clicking the `+ Create` button in the "Connected accounts" section.
```

File: app/furniture/marketplace/docs/README.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+  - You might name this account something like `<Space_Name> Dev`
+  - [Learn more](https://stripe.com/docs/connect)
+1. Follow instructions for creating a `Standard account`
+1. Once your new Connected account is setup, go back into the app, visit `http://convene.local:3000/spaces/<space_name>/rooms/store/marketplaces/<furniture_id>/edit` and click the `Connect to Stripe` button
```
Comment (KellyAH) [inline app/furniture/marketplace/docs/README.md:129]: And then what? is there a success message? how do I know it worked? Do I try to buy an item from a marketplace and see the transaction in stripe?

File: app/furniture/marketplace/docs/README.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+1. Access the app locally through `convene.local:3000`, instead of `localhost`
+1. Optionally, add `convene.local` as the `branded_domain` to whatever Space you are using for testing, to shorten the URL you use to access it
+
+Next, add your secret API key:
+1. Click on the `Developers` button next to `Test mode` in the top right of your Stripe dasbhoard
+1. Find your `Secret key` and copy it
+1. Paste the key into `Api token` field on your Stripe Utility settings page
+  - `http://convene.local:3000/spaces/<space_name>/utilities/<utiltiy_id>/edit`
+
+Next, configure Stripe to forward events to a local obtain a webhook secret key via the stripe cli:
```
Comment (rosschapman) [inline app/furniture/marketplace/docs/README.md:110]:
```suggestion
Next, using the stripe cli, configure Stripe to forward events to your local environment by obtainining a webhook secret key:
```

File: app/furniture/marketplace/docs/README.md (hunk 5 above)
Comment (rosschapman) [reply app/furniture/marketplace/docs/README.md:114]: #1, yep good catch. 
#2, also good catch! Yes, the code should be an example of printing the value to the screen.

File: app/furniture/marketplace/docs/README.md (hunk 8 above)
Comment (rosschapman) [reply app/furniture/marketplace/docs/README.md:125]: Nice catch! Let's add the image, too!

File: app/furniture/marketplace/docs/README.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+  - E.g. from view: `<%= debug polymorphic_url(marketplace.location(child: :stripe_events)) %>`
+  - [Read more](https://stripe.com/docs/webhooks/test#webhook-test-cli)
+1. Set stripe event forwarding to this url by executing: `$ stripe listen --forward-to <YOUR_FORWARD_TO_URL>`
+1. Copy the webhook signing secret from the output. The output from the above command should look something like:
+```
+> Ready! You are using Stripe API Version [2022-11-15]. Your webhook signing secret is whsec_***** (^C to quit)
+```
+1. Update the marketplace settings directly in the database by adding the `stripe_webhook_endpoint_secret` to `public.furnitures.settings`
+
+Finally, let's connect!
```
Comment (rosschapman) [inline app/furniture/marketplace/docs/README.md:123]:
```suggestion
Finally, let's setup a Standard Connected account in Stripe and then connect that account to the Stripe utility in our locally running application  
```

File: app/furniture/marketplace/docs/README.md (hunk 7 above)
Comment (rosschapman) [reply app/furniture/marketplace/docs/README.md:123]: Right, that's not very explanatory. How about [this](https://github.com/zinc-collective/convene/pull/1604/files#r1245669457 ) instead.

File: app/furniture/marketplace/docs/README.md (hunk 6 above)
Comment (rosschapman) [reply app/furniture/marketplace/docs/README.md:128]: "follow the UI flow prompts" is the language I was looking for. Nice!

File: app/furniture/marketplace/docs/README.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+1. First, create a Connected account within Stripe by visiting [https://dashboard.stripe.com/test/connect/accounts/overview](https://dashboard.stripe.com/test/connect/accounts/overview) and clicking the `+ Create` button in the "Connected accounts" section.
+  - You might name this account something like `<Space_Name> Dev`
+  - [Learn more](https://stripe.com/docs/connect)
+1. Follow instructions for creating a `Standard account`
```
Comment (rosschapman) [inline app/furniture/marketplace/docs/README.md:128]:
```suggestion
1. Follow the UI prompost for creating a `Standard account` 
```

**Comment by rosschapman** [comment]
Feeling appreciative of all the feedback! 

@anaulin @zspencer normally I would separate the commits sequentially to make reading the diffs easier. Good reminder. I've now pulled out the local development section from the primary README as @zspencer suggested. I like that. It also diffs better. 

@KellyAH seems like you got your local env working well enough to step through the new instructions!? Awesome QA strategy, and very helpful suggestions, indeed. I believe I've responded to all of them. Let me know if you want to keep looping on anything. Also since its docs, I'll just merge and we can keep refining as needed.

@zspencer I hear you on these:
> What parts of the docs are likely to go "out of date" by virtue of us simply improving our UX? (For example, setting the webhook secret via the UI vs via the console)
>
> What parts are duplicious of Stripe (For example, could the "turn on Test Mode" could be a link to https://stripe.com/docs/test-mode?)
The eternal conundrum. Good docs certainly increase maintenance burden. Let's see how quickly these go stale? And how much appetite the team has for maintenance work? I'm still learning how to be in the slipstream of this open source work and where to get the best bang for the buck. 

Also, I'm ok with trading on some overlap (added dependency), so a cow path exists; even if narrow shared (coupled) with Stripe. If again, you'll allow a little wait-and-see.

**Review by anaulin — APPROVED** [review]
This is a great improvement, thanks Kelly for the detailed feedback/QA on this, and Ross of course for making the improvements in the first place and then graciously handling all the feedback. 😻

Let's merge and then iterate if/when needed. 🚢

**Comment by zspencer** [comment]
dwight-yes

**Pull request merged**